src/ucp/core/ucp_am.c

```diff
@@ -1369,7 +1369,8 @@ ucp_am_handle_unfinished(ucp_worker_h worker, ucp_recv_desc_t *first_rdesc,
     am_id           = hdr->am_id;
     user_hdr_length = hdr->header_length;
     total_size      = first_ftr->total_size;
-    user_hdr        = UCS_PTR_BYTE_OFFSET(payload, total_size);
+    user_hdr        = (user_hdr_length > 0) ? UCS_PTR_BYTE_OFFSET(payload, total_size) :
```

Comment:
according to the doc it is undefined if the length is zero
```
 * @param [in]  header        User defined active message header. Can be NULL.
 * @param [in]  header_length Active message header length in bytes. If this
 *                            value is 0, the @a header pointer is undefined
 *                            and should not be accessed.
 ```
Maybe better fix the places where it is accessed when the length is 0?